File: Low Level Design (LLD)/DesignPatterns/3. Structural/Template/Cpp/main.cpp
#include "include/CSVDataParser.h"
#include "include/XMLDataParser.h"

int main() {
    std::cout << "---- Parsing CSV ----" << std::endl;
    CSVDataParser csv;
    csv.parseData();

    std::cout << std::endl;

    std::cout << "---- Parsing XML ----" << std::endl;
    XMLDataParser xml;
    xml.parseData();

    return 0;
}

File: Low Level Design (LLD)/DesignPatterns/3. Structural/Template/Java/Main.java
public class Main {
    public static void main(String[] args) {
        DataParser csvParser = new CSVDataParser();
        System.out.println("---- Parsing CSV ----");
        csvParser.parseData();

        System.out.println();

        DataParser xmlParser = new XMLDataParser();
        System.out.println("---- Parsing XML ----");
        xmlParser.parseData();
    }
}
